source/extensions/filters/udp/udp_proxy/udp_proxy_filter.cc

```diff
@@ -171,6 +174,33 @@ UdpProxyFilter::ActiveSession::ActiveSession(ClusterInfo& cluster,
       .connections()
       .inc();
 
+  if (use_original_src_ip_) {
+    ENVOY_LOG(debug, "The original src is enabled for address {}.",
+              addresses_.peer_->asStringView());
+
+    const auto host_addr = host_->address()->ip();
+    const int trans_value = 1;
+    Api::SysCallIntResult result;
+
+    if (host_addr->version() == Network::Address::IpVersion::v6) {
+      // set ipv6 option
+      result = Network::SocketOptionImpl::setSocketOption(*socket_, ENVOY_SOCKET_IPV6_TRANSPARENT,
+                                                          &trans_value, sizeof(trans_value));
+      RELEASE_ASSERT(!SOCKET_FAILURE(result.rc_), "");
+      if (!host_addr->ipv6()->v6only()) {
+        // set ipv4 option
+        result = Network::SocketOptionImpl::setSocketOption(*socket_, ENVOY_SOCKET_IP_TRANSPARENT,
+                                                            &trans_value, sizeof(trans_value));
+        RELEASE_ASSERT(!SOCKET_FAILURE(result.rc_), "");
```

Comment:
This is not guaranteed to be implemented on all platforms I don't think, so we should gracefully handle this being not suspported, and optimally fail during config load if possible.

Reply:
Ok. I will also address this.